app/assets/javascripts/discourse/app/components/sidebar/api-sections.gjs

```diff
@@ -1,32 +1,89 @@
 import Component from "@glimmer/component";
+import { cached } from "@glimmer/tracking";
+import { getOwner, setOwner } from "@ember/owner";
 import { service } from "@ember/service";
-import AdminHeader from "./admin-header";
 import ApiSection from "./api-section";
-import FilterNoResults from "./filter-no-results";
+import PanelHeader from "./panel-header";
 
 export default class SidebarApiSections extends Component {
   @service sidebarState;
 
   get sections() {
+    let sectionConfigs;
+
     if (this.sidebarState.combinedMode) {
-      return this.sidebarState.panels
+      sectionConfigs = this.sidebarState.panels
         .filter((panel) => !panel.hidden)
         .flatMap((panel) => panel.sections);
     } else {
-      return this.sidebarState.currentPanel.sections;
+      sectionConfigs = this.sidebarState.currentPanel.sections;
     }
+
+    return sectionConfigs.map((Section) => {
+      const SidebarSection = prepareSidebarSectionClass(Section);
+
+      const sectionInstance = new SidebarSection({
+        filterable:
+          !this.sidebarState.combinedMode &&
+          this.sidebarState.currentPanel.filterable,
+        sidebarState: this.sidebarState,
+      });
+
+      setOwner(sectionInstance, getOwner(this));
+
+      return sectionInstance;
+    });
+  }
+
+  get filteredSections() {
+    return this.sections.filter((section) => section.filtered);
   }
 
   <template>
-    <AdminHeader />
+    <PanelHeader @sections={{this.filteredSections}} />
 
-    {{#each this.sections as |sectionConfig|}}
-      <ApiSection
-        @sectionConfig={{sectionConfig}}
-        @collapsable={{@collapsable}}
-      />
+    {{#each this.filteredSections as |section|}}
+      <ApiSection @section={{section}} @collapsable={{@collapsable}} />
     {{/each}}
-
-    <FilterNoResults />
   </template>
 }
+
+// extends the class provided for the section to add functionality we don't want to be overridable when defining custom
+// sections using the plugin API, like for example the filtering capabilities
+function prepareSidebarSectionClass(Section) {
```

Comment:
This is because JS does not support marking a method as `final` in the base class